opentelemetry-api-shaded-for-instrumenting/build.gradle.kts

```diff
@@ -2,18 +2,46 @@ plugins {
   id("com.github.johnrengelman.shadow")
 
   id("otel.java-conventions")
+  id("otel.publish-conventions")
+}
+
+description = "opentelemetry-api shaded for internal javaagent usage"
+group = "io.opentelemetry.javaagent"
+
+val shadowInclude by configurations.creating {
+  isCanBeResolved = true
+  isCanBeConsumed = false
 }
 
 dependencies {
-  implementation("io.opentelemetry:opentelemetry-api")
-  implementation("io.opentelemetry:opentelemetry-api-metrics")
+  shadowInclude("io.opentelemetry:opentelemetry-api")
+  shadowInclude("io.opentelemetry:opentelemetry-api-metrics")
 }
 
 // OpenTelemetry API shaded so that it can be used in instrumentation of OpenTelemetry API itself,
 // and then its usage can be unshaded after OpenTelemetry API is shaded
 // (see more explanation in opentelemetry-api-1.0.gradle)
 tasks {
+  jar {
+    enabled = false
+  }
+
   shadowJar {
+    configurations = listOf(shadowInclude)
+    archiveClassifier.set("")
     relocate("io.opentelemetry", "application.io.opentelemetry")
   }
+
+  publishing {
+    publications {
+      named<MavenPublication>("maven") {
+        // shadow does not use default configurations
+        project.shadow.component(this)
+        // force jar packaging
+        pom {
+          packaging = "jar"
+        }
```

Comment:
I found a hack that fixes pom packaging issues 🙈  😂

Reply:
Would it be worth letting muzzle configuration specify a classifier instead of having to do this?

Reply:
> I found a hack that fixes pom packaging issues 🙈 😂

wow. just wow. 😂

> Would it be worth letting muzzle configuration specify a classifier instead of having to do this?

I do prefer not having an unnecessary classifier on shaded artifacts, though very flexible for "anything that works" when it comes to gradle shadow plugin. How I miss maven-shade-plugin...

Reply:
Yeah for these that aren't meant for end users to consume, I would be ok on going with simpler build scripts if it makes sense. For end user artifacts, no classifier is very worth it though

Reply:
Okay, I simplified the shadow build and added classifier property to muzzle check plugin. That should work for these internal artifacts.